src/factories/AnimationTimerFactory.java

```diff
@@ -48,53 +56,118 @@ public AnimationTimer makeAnimationTimer(GraphicsContext gc) {
 
     return new AnimationTimer() {
       public void handle(long currentNTime) {
+      	
+      	EndBoss endBoss = MainScreenController.getEndBoss();
+      	BoundingBox endBossbb = endBoss.getSprite().getBoundingBox();
+      	
+      	Lance lance = MainScreenController.getLance();
+      	BoundingBox lancebb = lance.getSprite().getBoundingBox();
+      	
+      	PlayerFish playerFish = MainScreenController.playerFish;
+      	Sprite pfSprite = playerFish.getSprite();
+      	BoundingBox pfbb = pfSprite.getBoundingBox();
 
         if (MainScreenController.playerHasWon()) {
           this.stop();
           Game.switchScreen("FXML/WinningScreen.fxml");
           Game.getMediaPlayer().stop();
           Game.getLogger().logSwitchScreen("WinningScreen");
+          
+          MainScreenController.setBossMode(false);
+          playerFish.setHasLance(false);
+          
+          endBossbb.setX(-2000);
+        	endBossbb.setY(-2000);
+        }
+        
+        if (pfbb.getWidth() > 60) {
```

Comment:
So we just have to change this number, then we can have a longer game, right?

< review